### PR TITLE
replace ha-icon_button and ha-icon in stack card editor

### PR DESCRIPTION
**Pull request by Misiu**
## Proposed change

I've replaced `ha-icon` and `ha-icon-button` with `mwc` equivalents.
I had to change `_handleMove` because when the listener was added to the button it fired for both button and icon click.
## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [x] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml

```

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue: https://github.com/home-assistant/frontend/issues/6139
- Link to documentation pull request:

## Checklist

- [x] The code change is tested and works locally.
- [x] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

File: src/panels/lovelace/editor/config-elements/hui-stack-card-editor.ts [hunk 1]
```diff
@@ -107,26 +107,29 @@ export class HuiStackCardEditor extends LitElement
...
-                  <ha-icon-button
-                    id="move-after"
-                    title="Move card after"
-                    icon="hass:arrow-right"
+                  <mwc-icon-button
+                    label="Move card after"
                     .disabled=${selected === numcards - 1}
-                    @click=${this._handleMove}
-                  ></ha-icon-button>
+                    @click=${() => this._handleMove(+1)}
```
Comment (zsarnett) [inline src/panels/lovelace/editor/config-elements/hui-stack-card-editor.ts:122]: You should just make this a property on the icon-button

Comment (zsarnett) [reply src/panels/lovelace/editor/config-elements/hui-stack-card-editor.ts:122]:
```suggestion
					.move=${1}
                    @click=${this._handleMove}
```

And then handle this in the function

File: src/panels/lovelace/editor/config-elements/hui-stack-card-editor.ts [hunk 2]
```diff
@@ -193,12 +196,12 @@ export class HuiStackCardEditor extends LitElement
     fireEvent(this, "config-changed", { config: this._config });
   }
 
-  private _handleMove(ev) {
+  private _handleMove(direction: number) {
     if (!this._config) {
       return;
     }
     const source = this._selectedCard;
-    const target = ev.target.id === "move-before" ? source - 1 : source + 1;
```
Comment (bramkragten) [inline src/panels/lovelace/editor/config-elements/hui-stack-card-editor.ts:201]: If you change `ev.target` to `ev.currentTarget` the old code will just work.

Comment (zsarnett) [reply src/panels/lovelace/editor/config-elements/hui-stack-card-editor.ts:201]: The old code is not good though :)

Comment (zsarnett) [reply src/panels/lovelace/editor/config-elements/hui-stack-card-editor.ts:201]: string comparison vs just adding the 1/-1 to a property is probably better

Comment (bramkragten) [reply src/panels/lovelace/editor/config-elements/hui-stack-card-editor.ts:201]: yes, but that still requires `currentTarget` :smile:

Comment (zsarnett) [reply src/panels/lovelace/editor/config-elements/hui-stack-card-editor.ts:201]: true

Comment (Misiu) [reply src/panels/lovelace/editor/config-elements/hui-stack-card-editor.ts:201]: ok, so `currentTarget` in function and a `.move` property right?

Comment (Misiu) [reply src/panels/lovelace/editor/config-elements/hui-stack-card-editor.ts:201]: done.
@zsarnett is there a way to show a tooltip for the `mwc-icon-button`? I've added labels, but it isn't showing. The icons are self-explanatory, but a tooltip would be a nice addition.
There is [an issue](https://github.com/material-components/material-components-web-components/issues/1499) in MWC repo, but sadly the status of the tooltip component is TBD.

Comment (bramkragten) [reply src/panels/lovelace/editor/config-elements/hui-stack-card-editor.ts:201]: use the `title` attribute

File: src/panels/lovelace/editor/config-elements/hui-stack-card-editor.ts [hunk 3]
```diff
@@ -107,26 +107,31 @@ export class HuiStackCardEditor extends LitElement
...
                     )}
                   </mwc-button>
-                  <ha-icon-button
-                    id="move-before"
-                    title="Move card before"
-                    icon="hass:arrow-left"
+
+                  <mwc-icon-button
                     .disabled=${selected === 0}
+                    label="Move card before"
```
Comment (bramkragten) [inline src/panels/lovelace/editor/config-elements/hui-stack-card-editor.ts:113]: While at it, can we localize these?

**Review by bramkragten — APPROVED** [review]

File: src/panels/lovelace/editor/config-elements/hui-stack-card-editor.ts (hunk 3 above)
Comment (Misiu) [reply src/panels/lovelace/editor/config-elements/hui-stack-card-editor.ts:113]: I was just going to ask should I do it 🙂
Sure, give me a second.

Comment (Misiu) [reply src/panels/lovelace/editor/config-elements/hui-stack-card-editor.ts:113]: @bramkragten last question.
In en.json we have card written with first letter [upper ](https://github.com/home-assistant/frontend/blob/dev/src/translations/en.json#L2246)and [lower](https://github.com/home-assistant/frontend/blob/dev/src/translations/en.json#L2238).
I think this should be consistent.

File: src/panels/lovelace/editor/config-elements/hui-stack-card-editor.ts (hunk 2 above)
Comment (Misiu) [reply src/panels/lovelace/editor/config-elements/hui-stack-card-editor.ts:201]: done

**Review by bramkragten — APPROVED** [review]

File: src/panels/lovelace/editor/config-elements/hui-stack-card-editor.ts (hunk 1 above)
Comment (Misiu) [reply src/panels/lovelace/editor/config-elements/hui-stack-card-editor.ts]: Done

File: src/panels/lovelace/editor/config-elements/hui-stack-card-editor.ts (hunk 3 above)
Comment (bramkragten) [reply src/panels/lovelace/editor/config-elements/hui-stack-card-editor.ts]: It probably should, but in a sentence, it should definitely not be uppercase. In a short action format... yeah probably also not?

Comment (Misiu) [reply src/panels/lovelace/editor/config-elements/hui-stack-card-editor.ts]: Thanks for your clarification.
I'll try to optimize the `title` property in the next PR's. In many places, it is used without dot notation and `hass` is used without the exclamation point: https://github.com/home-assistant/frontend/blob/dev/src/panels/config/devices/ha-config-device-page.ts#L268

Comment (bramkragten) [reply src/panels/lovelace/editor/config-elements/hui-stack-card-editor.ts]: it should be without exclamation

Comment (Misiu) [reply src/panels/lovelace/editor/config-elements/hui-stack-card-editor.ts]: there are about 380 places that use an exclamation. Like here (https://github.com/home-assistant/frontend/blob/dev/src/panels/config/automation/condition/types/ha-automation-condition-time.ts): 
image
In the same class has is defined as:
`@property({ attribute: false }) public hass!: HomeAssistant;`

So all those places should be changed? I'm confused 🤷‍♂️

Comment (bramkragten) [reply src/panels/lovelace/editor/config-elements/hui-stack-card-editor.ts]: Do you know how typescript works and what the exclamation means?

Comment (bramkragten) [reply src/panels/lovelace/editor/config-elements/hui-stack-card-editor.ts]: But if `hass` is defined as with a `!` then it is redundant in the rest of the file. Not that it matters for the actual output.

Comment (Misiu) [reply src/panels/lovelace/editor/config-elements/hui-stack-card-editor.ts]: @bramkragten I just checked to be sure that my thinking was correct. Nothing changed, it is still a not-null assertion operator (https://stackoverflow.com/questions/42273853/in-typescript-what-is-the-exclamation-mark-bang-operator-when-dereferenci)
So when `hass` is defined with `!` then `!` is redundant in labels and titles.
I'll pick a couple of files and will do a PR with the changes and we can discuss the change there.

**Pull request merged**